### DISCUSSION TITLE
Extending User model

### DISCUSSION THREAD

**Discussion by Jonston** [spatie/laravel-permission#2250] [Q&A]
I have following structure of Users.Student extends User and Teacher extends User.Those models has relevant roles (student, teacher).How can i take Student instance of Auth::user() if student is logged in?

**Comment by erikn69** [comment]
https://github.com/laravel/laravel/blob/8a8730c994849967db6fb493f524e42f66a05ab5/config/auth.php#L62-L66
You can add more providers on `config/auth.php`, also you can use multiple guards on the same file

**Reply by Jonston** [reply]
> https://github.com/laravel/laravel/blob/8a8730c994849967db6fb493f524e42f66a05ab5/config/auth.php#L62-L66 You can add more providers on `config/auth.php`, also you can use multiple guards on the same file


How to link this to the spatie/permissions package?can you show simple example?

**Reply by erikn69** [reply]
> How to link this to the spatie/permissions package?

 On the user logging, permission uses laravel's gate/auth

https://spatie.be/docs/laravel-permission/v5/basic-usage/multiple-guards

**Reply by Jonston** [reply]
@erikn69, Somethink like this?

```php
Route::get('login', function () {
    $email = \request('email');
    $password = \request('password');
    $user = App\User::where('email', $email)->first();

    if ( ! $user) {
        return response()->json([
            'message' => 'User not found'
        ], 404);
    }
    
    $guard = $user->role->name;

    Auth::guard($guard)->attempt(['email' => $email, 'password' => $password]);
});`

**Reply by erikn69** [reply]
> `$guard = $user->role->name;`

I don't know what returns `$user->role->name;`, but sure, you could use multiple guards

>Student extends User and Teacher extends User.

how you know when is Student or Teacher on that example?

**Reply by Jonston** [reply]
@erikn69 By role name.

**Reply by erikn69** [reply]
I think you mixed things up a bit, you forgot the extending, you left it pointless which was the question

**Reply by Jonston** [reply]
@erikn69 
I need the Auth::user() method to return the appropriate instance depending on what role the user has. If the role is a student, then new Student, if the role is a teacher, then new Teacher

**Reply by Jonston** [reply]
@erikn69 The issue is that students and teachers have different model relations. Students have assignments. Teachers have students, classromms, and so on.

**Reply by erikn69** [reply]
Ok
